app/domain/authentication/authn_jwt/get_restrictions_from_annotation.rb

```diff
@@ -0,0 +1,46 @@
+require 'command_class'
+
+module Authentication
+  module AuthnJwt
+    # Parses the given annotation name and tries to extract a restriction from it.
+    # A restriction in format "<Authenticator name>/<Service ID>/<Restriction name>"
+    #
+    # The first format is a general restriction.
+    # This means it is relevant to all services of this authenticator.
+    #
+    # The format is specific to the specified service ID.
+    # This means all other services will ignore it.
+    # Moreover, if both formats are found for the same restriction, the specific one will be used.
+    #
+    # This function returns the restriction name, if found (nil otherwise) and a
+    # boolean indicating if it is a general restriction or not.
+    GetRestrictionsFromAnnotation = CommandClass.new(
```

Comment:
Class name should reflect its meaning:
```suggestion
    GetServiceSpecificRestrictionsFromAnnotation = CommandClass.new(
```

Reply:
Remember to rename file name

Reply:
Done